lib/client/ClientAjax.js

```diff
@@ -0,0 +1,74 @@
+"use strict";
+
+var Promise = require("bluebird");
+var Backbone = require("backbone");
+var BootstrappedDataService = require("./BootstrappedDataService");
+var ClientTestable = require("./ClientTestable");
+var pathJoin = require("../util/pathJoin");
+var _ = require("underscore");
+
+var originalAjax = Backbone.ajax;
+
+var ClientAjax = {
+
+    setup: function(bootstrappedData, appRoot) {
+        ClientTestable.BackboneNativeAjax.Promise = Promise;
+
+        BootstrappedDataService.load(bootstrappedData);
+
+        Backbone.ajax = function(ajaxConfig) {
+            var dataInBootstrappedData = BootstrappedDataService.getFor(ajaxConfig);
+            var bubbledError;
+
+            function bubbleErrorFrom(prop) {
+                var callback = ajaxConfig[prop];
+
+                if (!_.isFunction(callback)) {
+                    return;
+                }
+
+                ajaxConfig[prop] = function() {
+                    try {
```

Comment:
I'm a little bit concerned about the try catch here. `try/catch` generally cause deopts, but I'm not sure how it would work inside the closure. It should be pulled out if possible.